src/main/java/org/jetbrains/research/intellijdeodorant/ide/ui/AbstractRefactoringPanel.java

```diff
@@ -137,6 +140,19 @@ private JScrollPane createTablePanel() {
         return scrollPane;
     }
 
+    /**
+     * Adds listener which invalidates found refactoring opportunities if Psi structure of project is changed.
```

Comment:
`Adds a listener that invalidates found refactoring opportunities it the structure of PSI was changed.`